### PR TITLE
Add FCIS coco demo, training and evaluation

### PR DESCRIPTION
**Pull request by knorth55**
FCIS coco train, demo, evaluation

### CODE REVIEW

File: examples/fcis/README.md [hunk 1]
```diff
@@ -7,29 +7,37 @@ SBD Train & Test
...
 ```
 
 ## Evaluation
-The evaluation can be conducted using [`chainercv/examples/instance_segmentation/eval_sbd.py`](https://github.com/chainer/chainercv/blob/master/examples/instance_segmentation)
+The evaluation for sbd dataset can be conducted using [`chainercv/examples/instance_segmentation/eval_sbd.py`](https://github.com/chainer/chainercv/blob/master/examples/instance_segmentation)
+and the one for coco dataset can be conducted using [`chainercv/examples/instance_segmentation/eval_coco.py`](https://github.com/chainer/chainercv/blob/master/examples/instance_segmentation)
 
 ## Train
 You can train the model with the following code.
 Note that this code requires `SciPy` module.
```
Comment (yuyu2172) [inline examples/fcis/README.md]: The README is missing `train_coco_multi.py`.

File: examples/fcis/demo.py [hunk 2]
```diff
@@ -15,12 +16,31 @@
...
-    model = FCISResNet101(
-        n_fg_class=20, pretrained_model=args.pretrained_model)
+    if args.dataset == 'sbd':
+        if args.pretrained_model is None:
+            args.pretrained_model = 'coco'
+        label_names = sbd_instance_segmentation_label_names
+        model = FCISResNet101(
+            n_fg_class=len(label_names),
+            pretrained_model=args.pretrained_model)
+    # coco
```
Comment (yuyu2172) [inline examples/fcis/demo.py]: I think this comment is not needed.

File: examples/fcis/demo.py [hunk 3]
```diff
@@ -15,12 +16,31 @@
...
+    parser.add_argument(
+        '--dataset', choices=('sbd', 'coco'), default='sbd')
     parser.add_argument('image')
     args = parser.parse_args()
 
-    model = FCISResNet101(
-        n_fg_class=20, pretrained_model=args.pretrained_model)
+    if args.dataset == 'sbd':
+        if args.pretrained_model is None:
+            args.pretrained_model = 'coco'
```
Comment (yuyu2172) [inline examples/fcis/demo.py]: Isn't this wrong?

File: examples/fcis/train_coco_multi.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    import ProposalTargetCreator
+from chainercv.extensions import InstanceSegmentationCOCOEvaluator
+from chainercv.links.model.ssd import GradientScaling
+
+from train_sbd import concat_examples
+from train_sbd import Transform
+
+
+def filter_dataset(dataset):
+    indices = []
```
Comment (yuyu2172) [inline examples/fcis/train_coco_multi.py]: How about using the public interface of slicableDataset?
`indices = np.array([i for i, label in enumerate(dataset.slice[:, ['label']]) if len(label) > 0], dtype=np.int32)`

File: examples/fcis/train_coco_multi.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    indices = chainermn.scatter_dataset(indices, comm, shuffle=True)
+    train_dataset = train_dataset.slice[indices]
+    train_iter = chainer.iterators.SerialIterator(train_dataset, batch_size=1)
+
+    # test dataset
+    if comm.rank == 0:
+        test_dataset = COCOInstanceSegmentationDataset(
+            year='2014', split='minival', use_crowded=True,
+            return_crowded=True, return_area=True)
+        test_dataset = filter_dataset(test_dataset)
```
Comment (yuyu2172) [inline examples/fcis/train_coco_multi.py]: We don't need to filter `test_dataset`.

File: examples/fcis/demo.py (hunk 3 above)
Comment (knorth55) [reply examples/fcis/demo.py]: Yes thanks

**Comment by yuyu2172** [comment]
I missed one more thing.
Could you add new pretrained models to the test below?
https://github.com/chainer/chainercv/blob/master/tests/experimental_tests/links_tests/model_tests/fcis_tests/test_fcis_resnet101.py#L86

**Comment by knorth55** [comment]
I updated to add test pretrained model
